arkouda/segarray.py

```diff
@@ -1542,23 +1422,9 @@ def register(self, user_defined_name):
         --------
         unregister, attach, is_registered
         """
-        try:
-            rep_msg = generic_msg(
-                cmd="register", args={"array": self.name, "user_name": user_defined_name}
-            )
-            if rep_msg != "success":
-                raise RegistrationError
-        except (
-            RuntimeError,
-            RegistrationError,
-        ):  # Registering two objects with the same name is not allowed
-            raise RegistrationError(f"Server was unable to register {user_defined_name}")
-
         self.name = user_defined_name
         self.segments.register(self.name + SEG_SUFFIX)
         self.values.register(self.name + VAL_SUFFIX)
-        self.lengths.register(self.name + LEN_SUFFIX)
-        self.grouping.register(self.name + GROUP_SUFFIX)
```

Comment:
Ah yeah. I meant to do that. However, now I am wondering if it makes more sense to register them or recompute. We should discuss this some offline and update with the direction we want to go.

Reply:
Yeah, working on updating this now.

Reply:
Ok I configured this so that `lengths` is registered. For the grouping, any access checks if the object is in the Symbol Table and if so returns, otherwise it is rebuilt. Similar workflow used for `_non_empty` as well to support the groupby.